tests/api_models/test_optional_fields.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional
+
+from zgw_consumers.api_models.base import Model, factory
+from zgw_consumers.api_models.catalogi import ZaakType
+from zgw_consumers.api_models.zaken import Zaak
+
+
+def test_absent_keys_for_optional_fields() -> None:
+    @dataclass
+    class TestModel(Model):
+        required_field: str
+        optional_field: Optional[datetime] = None
+
+    data = {"requiredField": "a value"}
+
+    instance = factory(TestModel, data)
+
+    assert instance.required_field == "a value"
+    assert instance.optional_field is None
+
+
+def test_absent_keys_for_zaaktype() -> None:
+    data = {
+        "aanleiding": "Voorbeeld",
+        "doel": "Voorbeeld",
+        "handelingBehandelaar": None,
+        "handelingInitiator": "Melden",
+        "identificatie": "6",
+        "indicatieInternOfExtern": "intern",
+        "omschrijving": "Aangeven",
+        "onderwerp": "Aangeven",
+        "url": "https://uniek-url",
+        "vertrouwelijkheidaanduiding": "openbaar",
+    }
+    instance = factory(ZaakType, data)
+
+    assert instance.omschrijving == "Aangeven"
+    assert instance.omschrijving_generiek is None
```

Comment:
Looked into this failing test and compared with the [API spec](https://catalogi-api.vng.cloud/api/v1/schema/#operation/zaaktype_read):

* `catalogus`is a required string field (format uri)
* `doorlooptijd` is a required string field (format duration)
* `opschorting_en_aanhouding_mogelijk` is a required boolean field
* `verlenging_mogelijk` is a required boolean field
* `publicatie_indicatie` is a required boolean field
* `producten_of_diensten` is a required list of strings (format uri)
* `besluittypen` is a required list of strings (format uri)
* `begin_geldigheid` is a required string field (format date)
* `versiedatum`  is a required string field (format date)

In short: the API provider is not standard-compliant here.

This test is useful, but should be adapted `with pytest.assertRaises(TypeError): ...` instead of the asserts.